pkg/helm/charts/tests/securedclusterservices/resources_test.go

```diff
@@ -0,0 +1,497 @@
+			isInitContainer: true,
+			values: []string{
+				defaultValues,
+				`
+scanner:
+  dbResources:
+    limits:
+      cpu: 100m
+      memory: 100Mi`,
+			},
+			expectedResources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("100m"),
+					corev1.ResourceMemory: resource.MustParse("100Mi"),
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		var tc = tc
+		s.T().Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			_, m := s.LoadAndRender(tc.values...)
+			require.NotEmpty(t, m)
+			p := s.ParseObjects(m)
+			objs := map[string]unstructured.Unstructured{}
+			for _, obj := range p {
+				objs[obj.GetKind()+"/"+obj.GetName()] = obj
+			}
+
+			var containers []corev1.Container
+			if tc.workloadKind == "Deployment" {
+				var deployment appsv1.Deployment
+				require.NoError(t, mustFindDeployment(objs, tc.workloadKind+"/"+tc.workloadName, &deployment))
+				if tc.isInitContainer {
+					containers = deployment.Spec.Template.Spec.InitContainers
+				} else {
+					containers = deployment.Spec.Template.Spec.Containers
+				}
+			} else if tc.workloadKind == "DaemonSet" {
+				var daemonSet appsv1.DaemonSet
+				require.NoError(t, mustFindDaemonSet(objs, tc.workloadKind+"/"+tc.workloadName, &daemonSet))
+				if tc.isInitContainer {
+					containers = daemonSet.Spec.Template.Spec.InitContainers
+				} else {
+					containers = daemonSet.Spec.Template.Spec.Containers
+				}
+			} else {
+				t.Fatalf("unsupported workload kind %s", tc.workloadKind)
+			}
+
+			var container corev1.Container
+			require.NoError(t, mustFindContainer(containers, tc.containerName, &container))
+			require.Truef(t, assertResourceRequirementsEqual(t, tc.expectedResources, container.Resources), "expected %v, got %v", tc.expectedResources, container.Resources)
+
+		})
+	}
+
+}
+
+func assertResourceRequirementsEqual(t *testing.T, expected, actual corev1.ResourceRequirements) bool {
+	return assertResourceListEqual(t, "limits", expected.Limits, actual.Limits) && assertResourceListEqual(t, "request", expected.Requests, actual.Requests)
+}
+
+func assertResourceListEqual(t *testing.T, typ string, expected, actual corev1.ResourceList) bool {
+	// check if keys are equal
+	for key := range expected {
+		if _, ok := actual[key]; !ok {
+			t.Errorf("expected key %s.%s not found in actual", typ, key)
+			return false
+		}
+	}
+	if len(expected) != len(actual) {
+		t.Errorf("expected %d keys, got %d in %s", len(expected), len(actual), typ)
+		return false
+	}
+	// check if values are equal
+	for resourceName, expectedValue := range expected {
+		actualValue := actual[resourceName]
+		if !expectedValue.Equal(actualValue) {
+			t.Errorf("expected value %v for key %s.%s, got %v", expectedValue, typ, resourceName, actualValue)
+			return false
+		}
+	}
+	return true
+}
+
+func mustFindDeployment(objs map[string]unstructured.Unstructured, name string, target *appsv1.Deployment) error {
```

Comment:
I think the convention is that `must` prefix means the function will panic if assertion fails.
I think the options here are:
1. drop the `must` prefix
2. keep the `must` prefix but change the signature to take a testing.T as the first argument, and return the target rather than take it as a parameter. Then do the require.NoError inside this function, instead of returning the error.